### DISCUSSION TITLE
Server component <Suspense /> not showing Instant Loading States when accessed through proxy

### DISCUSSION THREAD

**Discussion by cherealnice** [vercel/next.js#50377] [App Router]
Hello,

I'm currently building a new NextJS app from scratch using the app router. Currently, I am prefetching some data in my global layout, and I have a Suspense boundary around the data fetching component like this:

```
// app/layout.tsx

export default function Layout({ children }: { children: ReactNode }) {
  return (
    <html lang="en">
      <head />
      <body>
        <Providers>
          <Suspense fallback={<p>Loading data...</p>}>
            {/* @ts-expect-error Async Server Component */}
            <Prefetches>
                  <section>{children}</section>
            </Prefetches>
          </Suspense>
        </Providers>
      </body>
    </html>
  );
}
```

The problem is that when I access my app directly (http://localhost:3000), I do see the loading state, but when I access it through a proxy, I do not see the suspense boundary and I just get a blank page until the prefetches are complete.

I've done some digging and it looks like the requests to the next server (incl. headers) do not change when accessing through proxy. Is there something else that I need to configure on the proxy?

**Comment by cherealnice** [comment]
I think I figured it out, it seems like I need to have http streaming enabled

**Reply by ammar91** [reply]
@cherealnice  I'm running into a similar issue. I see you mentioned it may have to do with enabling the http streaming but not really get it. Can you please help me?

**Reply by cherealnice** [reply]
My problem stemmed from the fact that I was running a proxy in between the frontend and nextjs node server. Do you have a similar setup?

**Comment by ammar91** [comment]
yes

________________________________

**Comment by cherealnice** [comment]
In my proxy, I was reading the response to parse the content before sending it to the frontend. This caused the response to not be sent as a stream, which is how it should be sent with react server components. This article helped a lot https://blog.logrocket.com/streaming-ssr-with-react-18/

**Reply by frdteknikelektro** [reply]
Do you have any solution?

**Reply by cherealnice** [reply]
Yes, I changed my server code to not parse the response before forwarding it to the frontend app, so it was able to forward the content as a stream
